lib/pf/node.pm

```diff
@@ -1327,6 +1331,32 @@ sub _cleanup_attributes {
     $info->{'status'} = _cleanup_status_value($info->{'status'});
 }
 
+=item node_extended_data
+
+Get the node extended data
+
+=cut
+
+sub node_extended_data {
+    my ($mac) = @_;
+    my @namespaces = qw(mse);
+    if (@{$Config{advanced}{extended_profile_filter_namespaces}}) {
+        my %hash;
+        my $redis = pf::Redis->new(server => 'localhost:6379');
+        my $json = JSON::MaybeXS->new;
+        foreach my $namespace (@namespaces) {
+            my ($data) = $redis->get("extended:${namespace}:${mac}");
```

Comment:
IMO it should be the job of the 3rd party integration to know the state of the object it is storing.

So there two ways this could be done.

Different namespaces this way we can have separate expiration for each namespace.

set extended:MobileIron:00:11:22:33:44:55 '{ "allowed": 0 }' EX 1 # Expires in one second
set extended:OPSWAT:00:11:22:33:44:55 '{ "allowed": 1 }' EX 10 # Expires in 10 seconds
set extended:ZammitDM:00:11:22:33:44:55 '{ "allowed": 0 }'         # ZammitDM never expires

Or one namespace

set extended:sso:00:11:22:33:44:55 '{"MobileIron":0,"OPSWAT":1,"ZammitDM":0}'

Also my concern about using hashes it introduces more ways to get it wrong.

It introduces questions like the following
 * How do we decide when to serialize values using json vs just using plain text?
 * Which hash keys do we update and when?